### PR TITLE
ENH: Update bioformats library to version 5.9.0

### PR DESCRIPTION
**Pull request by blowekamp**

### CODE REVIEW

**Comment by blowekamp** [comment]
There are two problems with upgrading to the latest bioformats:
 - The downloaded JRE is the ancient 1.6 version and is not compatible with the current bioformats. I'm just setting JAVA_HOME locally, and in the CI to get around this right now
- There is the following error running the java code:
```
Exception in thread "main" java.lang.NoSuchMethodError: loci.formats.meta.IMetadata.getPixelsPhysicalSizeX(I)Lome/xml/model/primitives/PositiveFloat;
	at io.scif.itk.SCIFIOITKBridge.readImageInfo(SCIFIOITKBridge.java:303)
	at io.scif.itk.SCIFIOITKBridge.executeCommand(SCIFIOITKBridge.java:134)
	at io.scif.itk.SCIFIOITKBridge.executeCommand(SCIFIOITKBridge.java:109)
	at io.scif.itk.SCIFIOITKBridge.waitForInput(SCIFIOITKBridge.java:84)
	at io.scif.itk.SCIFIOITKBridge.executeCommand(SCIFIOITKBridge.java:164)
	at io.scif.itk.SCIFIOITKBridge.main(SCIFIOITKBridge.java:696)
```

I'm not sure If this is an API change or a java version compatibility change.

**Comment by ctrueden** [comment]
The `NoSuchMethodError` signifies a skew between two Java libraries. I am guessing it will fix things if we also upgrade `scifio-itk-bridge` to the matching Bio-Formats version. Newer versions of Bio-Formats require Java 7, and some of the dependent libraries are starting to depend on Java 8 now, so best would be to update to Java 8 across the board if possible.

The easiest way to update `scifio-itk-bridge` will be to update the `pom-scijava` parent to the latest (currently 23.1.1). I am in the process of producing a newer pom-scijava parent that addresses a couple of concerns though, particularly a Bio-Formats-related issue with native-lib-loader 2.3.0. So we may want to either: A) wait a few days here; or B) update `scifio-itk-bridge` but pin the `native-lib-loader` version at 2.2.0 for now, to avoid the issue.

**Comment by blowekamp** [comment]
Great! I think it will be a little less work if we just wait until the other work completes.

**Comment by blowekamp** [comment]
What is the next step to update `scifio-itk-bridge` so we can update this module?

**Comment by blowekamp** [comment]
@ctrueden What still needs to be done to update `scifio-itk-bridge` to the new Bio-Formats with newer Java? Anything I could do to help?

**Comment by ctrueden** [comment]
[https://github.com/scifio/scifio-imageio/pull/51#issuecomment-423215994](https://trello.com/c/WOBamsod/3-https-githubcom-scifio-scifio-imageio-pull-51issuecomment-423215994)

**Comment by ctrueden** [comment]
@blowekamp Apologies for the delay in reply. I have updated `scifio-itk-bridge` and [released a new stable version 1.2.1](https://travis-ci.org/scifio/scifio-itk-bridge/builds/431316654), which is built on Bio-Formats 5.9.2, and requires Java 8. Could you please give it a try and let me know how it goes?

**Comment by blowekamp** [comment]
I don't see that version available on [maven.imagej](https://mvnrepository.com/artifact/io.scif/scifio-itk-bridge?repo=imagej-releases), nor do I see a generate jar for downloaded on the linked travis build.

Naively running `man package` in a checkout out scifio-itk-bridge produced the following error:
```
ownloaded: https://repo.maven.apache.org/maven2/junit/junit/4.11/junit-4.11.jar (240 KB at 13294.2 KB/sec)
Downloaded: https://repo.maven.apache.org/maven2/org/codehaus/plexus/plexus-io/2.0.4/plexus-io-2.0.4.jar (57 KB at 2845.7 KB/sec)
Downloaded: https://repo.maven.apache.org/maven2/com/ibm/icu/icu4j/56.1/icu4j-56.1.jar (10731 KB at 41431.8 KB/sec)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 14.124s
[INFO] Finished at: Fri Sep 21 10:31:21 EDT 2018
[INFO] Final Memory: 14M/1891M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce (enforce-rules) on project scifio-itk-bridge: Unable to parse configuration of mojo org.apache.maven.plugins:maven-enforcer-plugin:1.4.1:enforce for parameter ignoreClasses: Cannot assign configuration entry 'ignoreClasses' with value 'org.hibernate.stat.ConcurrentStatisticsImpl,org.junit.runner.Runner' of type java.lang.String to property of type java.lang.String[] -> [Help 1]
```
```
$ java -version; mvn -version
openjdk version "1.8.0_181"
OpenJDK Runtime Environment (build 1.8.0_181-b13)
OpenJDK 64-Bit Server VM (build 25.181-b13, mixed mode)
Apache Maven 3.0.5 (Red Hat 3.0.5-17)
Maven home: /usr/share/maven
Java version: 1.7.0_191, vendor: Oracle Corporation
Java home: /usr/lib/jvm/java-1.7.0-openjdk-1.7.0.191-2.6.15.4.el7_5.x86_64/jre
Default locale: en_US, platform encoding: UTF-8
OS name: "linux", version: "3.10.0-693.17.1.el7.x86_64", arch: "amd64", family: "unix"
```

**Comment by ctrueden** [comment]
@blowekamp The `scifio-itk-bridge-1.2.1.jar` artifact can be downloaded from `maven.imagej.net` [here](https://maven.imagej.net/service/local/artifact/maven/redirect?r=releases&g=io.scif&a=scifio-itk-bridge&v=1.2.1&e=jar). There should be no need to rebuild it.

If there is build machinery (e.g. on the CMake side) that wants to download this artifact from somewhere, then [this link](http://maven.imagej.net/content/groups/public/io/scif/scifio-itk-bridge/1.2.1/scifio-itk-bridge-1.2.1.jar) might be best, since it should be future-proof.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -19,7 +19,7 @@
 # for OS 10.9.
 #
 set( SCIFIOJars
-  https://downloads.openmicroscopy.org/bio-formats/5.0.4/artifacts/bioformats_package.jar
+  https://downloads.openmicroscopy.org/bio-formats/5.8.2/artifacts/bioformats_package.jar
   https://maven.imagej.net/service/local/repositories/releases/content/io/scif/scifio-itk-bridge/1.2.0/scifio-itk-bridge-1.2.0.jar
```
Comment (ctrueden) [inline src/CMakeLists.txt]: I think it should be enough to: A) change the Bio-Formats version here to `5.9.2`, and B) change the scifio-itk-bridge version here to `1.2.1`.

Please test and let me know how it goes! If it works, amend this commit, and I'll merge it.

**Comment by blowekamp** [comment]
Great! Thanks for the URL links!

CircleCI provides a system Java JRE and sets JAVA_HOME to use it to override the old JAVA downloaded version.

@thewtex Is it really necessary to provide a JAVA distribution? Can we just rely on a system JAVA installed? We could add some better CMake or run-time checking.

**Comment by thewtex** [comment]
> @thewtex Is it really necessary to provide a JAVA distribution? Can we just rely on a system JAVA installed? We could add some better CMake or run-time checking.

Java is typically not available by default on Linux or Windows systems. Installation is difficult enough that we should try to make it work out of the box, if possible.

**Comment by blowekamp** [comment]
Having an old Java version can be considered a significant security concern by some, even in an isolated use case like this. It would be quite a burden to keep the bundled version up today with the latest security patches.

Installation on Linux is generally trivial with the system package managers. 

There is current no supplied solution for OS X. 

I could see windows being more problematic, but with third party package managers like chocolaty  the burden should be reasonable. 

Effort would need to be put into detecting if the version of Java is compatible and improve error messages and documentation when there are problem.

**Comment by thewtex** [comment]
Shifting the burden to users puts the burden on many more people.

Keep in mind that not everyone has administrator access on their Linux system. Also, most people do not have Chocolatey and few are interested in installing it. In the end, we want this to not be used just by a few developers, but we want it to be incorporated into end-user applications. These end users will just walk away if they see the command line instructions to install a developer package manager or go to Oracle's website and pick the right Java runtime for their architecture. This is why Fiji ships with a Java runtime.

**Pull request merged**